### ISSUE TITLE
Oversmoothed Texture + Geometric Discrepancies

### ISSUE THREAD

**Issue by AvirupJU** [adobe-research/diffusion-rig#12]
Hi, I fine-tuned the model on an image set of Obama but the results in a self-reconstruction setting i.e. source = target are subpar. The texture is oversmoothed. Even for expression transfer the results are not accurate.<br>
Any idea why this might be happening? Is it a mismatch bw data distributions (which should not be the case since we are finetuning the model), or some hyperparameter issue?
[Image order: source, self-recons, target, expression from target]
![image](https://github.com/user-attachments/assets/45798f78-68e4-4338-a125-ca66c35a5d1f)
![image](https://github.com/user-attachments/assets/4c0ec93e-9d8d-45b7-8b43-6a062a81d349)
![image](https://github.com/user-attachments/assets/d82403b7-a490-4127-a7f3-8a930aac1c96)
![image](https://github.com/user-attachments/assets/b2e8293d-853a-4ab9-802b-9ef33c908150)

**Comment by zh-ding** [comment]
If you could provide the subset you're using, I should be able to take a look and try to help. Btw, did you follow the alignment process to align the images first?

**Comment by AvirupJU** [comment]
[Obama.zip](https://github.com/user-attachments/files/17644558/Obama.zip)
[Macron.zip](https://github.com/user-attachments/files/17645797/Macron.zip)

These are the aligned images. I noticed that the texture problem arose due to alignment issues and I have resolved that. Also, the mode should have been ```exp```. However, the expression transfer is still <i>weird</i> in certain cases. Here's the script for reference:
```
python scripts/inference.py --source per_aligned/Obama/image_0002.png \
                            --target per_aligned/Obama/image_0008.png \
                            --output_dir inference \
                            --modes expression \
                            --model_path log/stage2/model005000.pt \
                            --meanshape "obama_deca.lmdb/mean_shape.pkl" \
                            --timestep_respacing ddim20
```
Here are the results: [src, target, transferred]
<p align="center">
  <img src="https://github.com/user-attachments/assets/620a6f5c-4a97-4efc-ab19-c4ee02e8e6c8" alt="Source Image" width="30%">
  <img src="https://github.com/user-attachments/assets/6e05c8a8-fc37-4165-9bed-fc53f1a002a3" alt="Target Image" width="30%">
  <img src="https://github.com/user-attachments/assets/c287c09b-2456-4452-873d-c369fd34ffc7" alt="Transferred Image" width="30%">
</p>
For Macron it is replicating the source directly:
<p align="center">
  <img src="https://github.com/user-attachments/assets/ebd8c3bb-a793-4267-b6b4-c9cd1ff5458e" alt="Image 1" width="30%">
  <img src="https://github.com/user-attachments/assets/e0205efa-1d2b-4bda-bd0d-8532e7f7c63e" alt="Image 2" width="30%">
  <img src="https://github.com/user-attachments/assets/83482343-8d3c-46a5-a4b4-6ac83925752a" alt="Image 3" width="30%">
</p>
But in another instance, where the target pose is closer to the source pose the model gives a better <i>approximation</i>, which looks like a linear combination of the two expressions
<p align="center">
  <img src="https://github.com/user-attachments/assets/ebd8c3bb-a793-4267-b6b4-c9cd1ff5458e" alt="Image 1" width="30%">
  <img src="https://github.com/user-attachments/assets/6379c3a8-056f-446b-9471-1b5fdd620c54" alt="Image 2" width="30%">
  <img src="https://github.com/user-attachments/assets/df072cdf-a785-4aaf-b2ee-60ca42d189db" alt="Image 3" width="30%">
</p>
<br>
I suppose the expression is not quite disentangled from the pose. Would love to know what you think!

**Comment by AvirupJU** [comment]
Fixed it. The mode should be ```exp```, not ```expression```

**Comment by zh-ding** [comment]
Glad to hear that the texture problem has been resolved after performing the image alignment process. 

Regarding the expression problem in certain cases after you used the correct mode, it's a bit hard to obtain super accurate expression transfer since we replied on DECA to estimate the expression therefore limited by DECA's estimation capability as we discussed in the limitation part of the paper. One might try to replace DECA with other models to enhance the accuracy of the control. I would also suggest rendering albedos/normals using the FLAME parameters estimated by DECA to have a better view of the issue.

**Comment by AvirupJU** [comment]
Yeah, I can try that. [DIffusion Avatars](https://tobias-kirschstein.github.io/diffusion-avatars/) does something similar with albedos and normal maps, the only difference being the underlying 3DMM. Also, some of its precision can be attributed to the subject-specific approach I guess. But anyway, thanks for the feedback!

**Comment by AvirupJU** [comment]
Hi, I figured that the issue is correlated with the diversity of the album images. If the images are perceptually similar the model learns an entangled representation where expressions, shape, appearance are tied up. In my case, I was using some frames from a talking face video. This is indeed very interesting! Any idea what's happening underneath?
Cheers!
